plugins/processors/port_name/port_name.go

```diff
@@ -0,0 +1,151 @@
+package portname
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+var sampleConfig = `
+[[processors.port_name]]
+  ## Name of tag holding the port number
+  # tag = "port"
+
+  ## Name of output tag where service name will be added
+  # dest = "service"
+
+  ## Default tcp or udp
+  # default_protocol = "tcp"
+`
+
+type sMap map[string]map[int]string // "https" == services["tcp"][443]
+
+var services sMap
+
+type PortName struct {
+	SourceTag       string `toml:"tag"`
+	DestTag         string `toml:"dest"`
+	DefaultProtocol string `toml:"default_protocol"`
+}
+
+func (d *PortName) SampleConfig() string {
+	return sampleConfig
+}
+
+func (d *PortName) Description() string {
+	return "Filter metrics with repeating field values"
```

Comment:
Description needs updated

Reply:
Yes, I missed that when stealing this from dedup.  fixed